### ISSUE TITLE
failed to reproduce the result on REVERIE

### ISSUE THREAD

**Issue by qizhust** [YushengZhao/TD-STP#3]
Thanks for the code! After finetuning, I can get similar results as reported on R2R, yet the results on REVERIE are far from reported ones. Could you share the exact config file you used for the experiment?

**Comment by YushengZhao** [comment]
what's your results?

**Comment by qizhust** [comment]
Hi, I got the following result on val_unseen split. RGS and RGSPL is much lower.

|           | SR    | SPL   | OSR   | RGS   | RGSPL |
|-----------|-------|-------|-------|-------|-------|
| report    | 34.88 | 27.32 | 39.48 | 21.16 | 16.56 |
| reproduce | 34.71 | 28.83 | 36.69 | 18.97 | 15.75 |

**Comment by qizhust** [comment]
Besides, could you explain how did you implement the global action space in Section 3.3? According to the source code 
https://github.com/YushengZhao/TD-STP/blob/4ba1feb50f0bbc6f9608e79337e5195508739f0c/finetune_src/models/vilmodel_cmt.py#L758

It seems that the act_logits is still llimited to adjacent locations.

Thanks.

**Comment by YushengZhao** [comment]
Thank you for your attention to our work！
1. For RGS and RGSPL: 
As we explained in our paper: “Since our model focuses mainly on navigation, we use a ViL-BERT model fine-tuned on the REVERIE training set to perform the grounding task when navigation ends." In this repo, we only release the navigation part for REVERIE dataset, which can reach the same SR and SPL accuracy in our paper. The RGS and RGSPL maybe a little bit lower, since it is predicted by the navigation model itself. 
To achieve the RGS and RGSPL in the paper, please train a separated [ViL-BERT](https://github.com/facebookresearch/vilbert-multi-task) on the REVERIE training split and perform the referring part at the end of the navigation. Because the referring part is not the contribution of our paper and is easy to tune, we do not plan to release that part code. 
2. For the global action spaces:
please refer to the generation part of the ```ob_nav_types```, which considers the global action space.
https://github.com/YushengZhao/TD-STP/blob/main/finetune_src/r2r/agent_cmt.py#L106

**Comment by qizhust** [comment]
Sorry, I don't understand why the generation gives global action space. Could you explain it in detail?

https://github.com/YushengZhao/TD-STP/blob/4ba1feb50f0bbc6f9608e79337e5195508739f0c/finetune_src/r2r/agent_cmt.py#L106

it sets candidate viewpoints as 1, end as 2, and pad as 0.

**Comment by YushengZhao** [comment]
As indicated in the paper, the global action space includes the history tokens and view tokens, which are candidates for the next step. 

It is hard to explain all the details for the global action space, which is mainly implemented in the ```rollout``` function (https://github.com/YushengZhao/TD-STP/blob/main/finetune_src/r2r/agent_cmt.py#L414) and the functions it calls.
If you are interested, you can set some breakpoints at that function for better understanding.

**Comment by qizhust** [comment]
Got it, thanks!
